examples/mpi/lammps/Dockerfile

```diff
@@ -1,29 +1,41 @@
 # ch-test-scope: full
 FROM openmpi
-WORKDIR /
+WORKDIR /usr/local/src
 
 # Packages for building.
 RUN apt-get install -qy --no-install-recommends \
+    cmake \
     git \
     patch \
-    python-dev
+    python-dev \
+    python-pip \
+    python-setuptools 
 
+# Building mpi4py from source to ensure it is built against our MPI build
+# Building numpy from source to work around issues seen on Aarch64 systems
+RUN pip install --no-binary=mpi4py,numpy mpi4py numpy 
 # Build LAMMPS.
-ENV LAMMPS_VERSION 17Nov16
-ENV LAMMPS_DIR lammps-$LAMMPS_VERSION
-ENV LAMMPS_TAR $LAMMPS_DIR.tar.gz
-RUN wget -nv http://lammps.sandia.gov/tars/$LAMMPS_TAR
-RUN tar xf $LAMMPS_TAR
-RUN    cd $LAMMPS_DIR/src \
-    && python Make.py -j $(getconf _NPROCESSORS_ONLN) -p none \
-              std no-lib reax meam poems python reaxc orig -a lib-all mpi
-RUN    mv $LAMMPS_DIR/src/lmp_mpi /usr/bin \
-    && ln -s ./$LAMMPS_DIR /lammps
-RUN    cd $LAMMPS_DIR/python \
-    && python2.7 install.py
-
-# Make another test input file.
-# Patch instead of including in.large.melt in full because LAMMPS is GPL.
-COPY melt.patch /lammps/examples/melt
-RUN patch -p1 -o /lammps/examples/melt/in.large.melt \
-               < /lammps/examples/melt/melt.patch
+ENV LAMMPS_VERSION 19Sep19
+RUN wget -nv http://lammps.sandia.gov/tars/lammps-$LAMMPS_VERSION.tar.gz
+RUN tar xf lammps-$LAMMPS_VERSION.tar.gz
+RUN mkdir lammps-${LAMMPS_VERSION}.build
+RUN    cd lammps-${LAMMPS_VERSION}.build \
+    && cmake -DCMAKE_INSTALL_PREFIX=/usr \
+             -DCMAKE_BUILD_TYPE=Release \
+             -DBUILD_MPI=yes \
+             -DBUILD_LIB=on \
+             -DBUILD_SHARED_LIBS=on \
+             -DPKG_DIPOLE=yes \
+             -DPKG_KSPACE=yes \
+             -DPKG_POEMS=yes \
+             -DPKG_PYTHON=yes \
+             -DPKG_USER-REAXC=yes \
+             -DPKG_USER-MEAMC=yes \
+             -DLAMMPS_MACHINE=mpi \
+          ../lammps-${LAMMPS_VERSION}/cmake \
+    && make -j $(getconf _NPROCESSORS_ONLN) install
+RUN ln -s /usr/local/src/lammps-${LAMMPS_VERSION}/ /lammps
+# Patch in.melt instead of including in.large.melt because LAMMPS is GPL.
+# Patch simple.py to uncomment mpi4py calls and disable file output.
+COPY melt-and-simple.patch /lammps
+RUN patch -d / -p1 < /lammps/melt-and-simple.patch
```

Comment:
I'd suggest keeping two patches for the separate targets.

Reply:
Since there are three targets, in.melt, in.simple, and simple.py are you thinking a patch for the former and one patch for the second two? Melt.patch could be restored to what is in git master and a second could be created that patches in.simple and simple.py. A downside of this would be that the patches would be inconsistently doing similar things. The melt.patch creates a new file for its result and simple.patch would be an in-place patch.

Reply:
Per the meeting today we decided two separate patches, both in-place.